util/reggen/multi_register.py

```diff
@@ -140,6 +147,14 @@ def get_field_list(self) -> List[Field]:
     def is_homogeneous(self) -> bool:
         return self.reg.is_homogeneous()
 
+    def handle_async(self, clocks: Clocking) -> object:
```

Comment:
Like the `Register` code, I think this returns `Optional[ClockingItem]`?

Reply:
so dummy question here, what in general are the rules for when to explicitly call out the object name vs using `object`?
i ask only because lint didn't seem to flag this, and I'm wondering if it's supposed to do.

Reply:
Well, `object` is a super-type that allows any value at all (so doesn't really provide any information). You can always treat a value as type `object`, but the problem happens when you try to use the result of the function. If we ran MyPy on the call-site (which we don't, because it's in topgen, I think), we'd see it complain about trying to do "ClockingItem stuff" with something whose type it doesn't know.

My general approach is to use `object` when we genuinely don't know the type (for example, we've just read something from an hjson/yaml file) and refine to a useful type with checks (and error messages!). If MyPy isn't clever enough to follow some logic about why "`x` **must** be of type `T` because the code just did XYZ", I insert some assertions.